File: Backend/Travle.Model/Access/ResetPasswordRequest.cs
namespace Travle.Model.Access
{
    /// <summary>Completes the reset flow: the emailed code plus the new password.</summary>
    public class ResetPasswordRequest
    {
        public string Email { get; set; } = string.Empty;
        public string Code { get; set; } = string.Empty;
        public string NewPassword { get; set; } = string.Empty;
        public string ConfirmNewPassword { get; set; } = string.Empty;
    }
}

File: UI/travle_core/lib/src/models/reset_password_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'reset_password_request.g.dart';

/// Completes the reset flow: the emailed [code] plus the new password.
@JsonSerializable()
class ResetPasswordRequest {
  ResetPasswordRequest({
    required this.email,
    required this.code,
    required this.newPassword,
    required this.confirmNewPassword,
  });

  final String email;
  final String code;
  final String newPassword;
  final String confirmNewPassword;

  factory ResetPasswordRequest.fromJson(Map<String, dynamic> json) =>
      _$ResetPasswordRequestFromJson(json);

  Map<String, dynamic> toJson() => _$ResetPasswordRequestToJson(this);
}
